carma-messenger-core/cpp_message/src/MobilityPath_Message.cpp

```diff
@@ -0,0 +1,241 @@
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage02.body.location.ecefX=location_val;
+
+        location_val=plainMessage.trajectory.location.ecef_y;
+        if(location_val> LOCATION_MAX || location_val<LOCATION_MIN){
+            ROS_WARN_STREAM("Location ecefY is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage02.body.location.ecefY=location_val;
+
+        location_val=plainMessage.trajectory.location.ecef_z;
+        if(location_val> LOCATION_MAX_Z || location_val<LOCATION_MIN_Z){
+            ROS_WARN_STREAM("Location ecefX is out of range");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+        message->value.choice.TestMessage02.body.location.ecefZ=location_val;
+
+        //get location timestamp and convert to char array
+        uint64_t time=plainMessage.trajectory.location.timestamp;
+        std::string timestamp=std::to_string(time);
+        size_t string_size=timestamp.size();
+        uint8_t string_location_timestamp[string_size];
+        for(size_t i=0;i<string_size;i++)
+        {
+            string_location_timestamp[i]=timestamp[i];
+        }
+        message->value.choice.TestMessage02.body.location.timestamp.buf=string_location_timestamp;
+        message->value.choice.TestMessage02.body.location.timestamp.size=string_size;
+
+        //trajectory
+        auto offset_count=plainMessage.trajectory.offsets.size();
+        
+        if(offset_count>MAX_POINTS_IN_MESSAGE){
+            ROS_WARN_STREAM("offset count greater than 60.");
+            return boost::optional<std::vector<uint8_t>>{};
+        }
+
+        MobilityLocationOffsets* offsets_list;
+        offsets_list=(MobilityLocationOffsets*)calloc(1,sizeof(MobilityLocationOffsets));
+
+        MobilityECEFOffset* Offsets;    
+        for(int i=0;i<offset_count;i++){
+            Offsets=(MobilityECEFOffset*)calloc(1,sizeof(MobilityECEFOffset));
+            Offsets->offsetX=plainMessage.trajectory.offsets[i].offset_x;
+            Offsets->offsetY=plainMessage.trajectory.offsets[i].offset_y;
+            Offsets->offsetZ=plainMessage.trajectory.offsets[i].offset_z;
+            asn_sequence_add(&offsets_list->list,Offsets);
+        }
+
+        message->value.choice.TestMessage02.body.trajectory=*offsets_list;
+        
+        ec=uper_encode_to_buffer(&asn_DEF_MessageFrame,0, message, buffer, buffer_size);
+        //log a warning if it fails
+        if(ec.encoded==-1){
+            return boost::optional<std::vector<uint8_t>>{}; 
+        }
+
+        //copy to byte array msg
+        auto array_length=ec.encoded/8;
+        std::vector<uint8_t> b_array(array_length);
+        for(auto i = 0; i < array_length; i++) b_array[i] = buffer[i];
```

Comment:
use size_t.
Also add an empty line below this. Otherwise it gives the impression that the return statement is running in the for-loop

Reply:
Removed all instances of auto and added the empty line.